packages/beeq/src/components/empty-state/bq-empty-state.tsx

```diff
@@ -0,0 +1,117 @@
+import { validatePropValue } from '../../shared/utils';
+
+/**
+ * @part body - The container `<div>` that wraps the alert description content
+ * @part footer - The container `<div>` that wraps the alert footer content
+ * @part icon - The `<bq-icon>` element used to render a predefined icon size based on the empty state size (small, medium, large)
+ * @part title - The container `<div>` that wraps the empty state title content
+ * @part wrapper - The wrapper container `<div>` of the element inside the shadow DOM
+ */
+
+@Component({
+  tag: 'bq-empty-state',
+  styleUrl: './scss/bq-empty-state.scss',
+  shadow: true,
+})
+export class BqEmptyState {
+  // Own Properties
+  // ====================
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqEmptyStateElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  // Public Property API
+  // ========================
+
+  /** The size of the empty state component */
+  @Prop({ reflect: true, mutable: true }) size: TEmptyStateSize = 'medium';
+
+  // Prop lifecycle events
+  // =======================
+
+  @Watch('size')
+  checkPropValues() {
+    validatePropValue(EMPTY_STATE_SIZE, 'medium', this.el, 'size');
+  }
+
+  // Events section
+  // Requires JSDocs for public API documentation
+  // ==============================================
+
+  // Component lifecycle events
+  // Ordered by their natural call order
+  // =====================================
+
+  componentWillLoad() {
+    this.checkPropValues();
+  }
+
+  // Listeners
+  // ==============
+
+  // Public methods API
+  // These methods are exposed on the host element.
+  // Always use two lines.
+  // Public Methods must be async.
+  // Requires JSDocs for public API documentation.
+  // ===============================================
+
+  // Local methods
+  // Internal business logic.
+  // These methods cannot be called from the host element.
+  // =======================================================
+
+  private get iconSize(): number {
+    return SIZE_TO_VALUE_MAP[this.size] || SIZE_TO_VALUE_MAP.medium;
+  }
+
+  // render() function
+  // Always the last one in the class.
+  // ===================================
+
+  render() {
+    return (
+      <Host>
+        <div class="bq-empty-state" part="wrapper">
+          <div class={{ [`bq-empty-state-icon-margin-bottom__${this.size}`]: true }}>
+            <slot name="icon">
+              <bq-icon size={this.iconSize} name="database" part="icon" exportparts="base,svg" />
+            </slot>
+          </div>
+          <div
+            class={{
+              'title-font font-bold leading-regular text-text-primary': true,
+              [`bq-empty-state-title-font-size__${this.size}`]: true,
+              [`bq-empty-state-title-margin-bottom__${this.size}`]: true,
+            }}
+            part="title"
+          >
```

Comment:
I would avoid declaring CSS vars for `font-size` and `margin-bottom`, since consumers can override this easily on integration by wrapping the content inside a div or span container.

```suggestion
          <div
            class={{
              'title-font font-bold leading-regular text-text-primary': true,
              'text-s mb-m': this.size === 'small'
              'text-m mb-l': this.size === 'medium'
              'text-l mb-xl': this.size === 'large'
            }}
            part="title"
          >
```

Reply:
Keep in mind that margin-bottom should only applied if there's body content, otherwise the component takes  more space than it should

![CleanShot 2023-12-06 at 16 50 33](https://github.com/Endava/BEEQ/assets/328492/5cc81637-261a-4fd9-a633-d4fb01010ffe)